sktime/forecasting/model_selection/tests/test_tune.py

```diff
@@ -329,40 +330,100 @@ def test_skoptcv_multiple_forecaster():
     assert len(sscv.cv_results_) == 5
 
 
-@pytest.mark.xfail
+@pytest.fixture(scope="module")
+def optuna_param_grids():
+    if _check_soft_dependencies("optuna"):
+        import optuna
+
+        PIPE_GRID_OPTUNA = {
+            "transformer__forecaster__degree": optuna.distributions.IntDistribution(
+                1, 2
+            ),
+            "forecaster__strategy": optuna.distributions.CategoricalDistribution(
+                ["last", "mean"]
+            ),
+        }
+        NAIVE_GRID_OPTUNA = {
+            "window_length": optuna.distributions.IntDistribution(
+                *TEST_WINDOW_LENGTHS_INT
+            )
+        }
+
+        return {"NAIVE": NAIVE_GRID_OPTUNA, "PIPE": PIPE_GRID_OPTUNA}
+
+
+@pytest.fixture(scope="module")
+def optuna_samplers():
+    if _check_soft_dependencies("optuna"):
+        import optuna
+
+        return [
+            None,
+            optuna.samplers.NSGAIISampler(seed=42),
+            optuna.samplers.QMCSampler(seed=42),
+            optuna.samplers.CmaEsSampler(seed=42),
+        ]
+
+
+forecasters_optuna_test = {
+    "NAIVE": NAIVE,
+    "PIPE": PIPE,
+}
+
+
 @pytest.mark.skipif(
     not run_test_for_class(ForecastingOptunaSearchCV),
     reason="run test only if softdeps are present and incrementally (if requested)",
 )
-@pytest.mark.parametrize(
-    "forecaster, param_grid", [(NAIVE, NAIVE_GRID), (PIPE, PIPE_GRID)]
-)
 @pytest.mark.parametrize("scoring", TEST_METRICS)
 @pytest.mark.parametrize("error_score", ERROR_SCORES)
 @pytest.mark.parametrize("cv", CVs)
+@pytest.mark.parametrize(
+    "forecaster_key, grid_key", [("NAIVE", "NAIVE"), ("PIPE", "PIPE")]
+)
 @pytest.mark.parametrize("n_iter", TEST_N_ITERS)
-@pytest.mark.parametrize("random_state", TEST_RANDOM_SEEDS)
-def test_optuna(forecaster, param_grid, cv, scoring, error_score, n_iter, random_state):
+def test_optuna(
+    forecaster_key,
+    grid_key,  # This will be either "NAIVE" or "PIPE"
+    cv,
+    scoring,
+    error_score,
+    n_iter,
+    optuna_samplers,  # Fixture providing samplers
+    optuna_param_grids,  # Fixture providing parameter grids
+):
     """Test TuneForecastingOptunaCV.
 
     Tests that TuneForecastingOptunaCV successfully searches the parameter
     distributions to identify the best parameter set
     """
     y, X = load_longley()
-    rscv = ForecastingOptunaSearchCV(
-        forecaster,
-        param_grid=param_grid,
-        cv=cv,
-        scoring=scoring,
-        error_score=error_score,
-    )
-    rscv.fit(y, X)
-
-    param_distributions = list(
-        ParameterSampler(param_grid, n_iter, random_state=random_state)
-    )
-    _check_cv(forecaster, rscv, cv, param_distributions, y, X, scoring)
-    _check_fitted_params_keys(rscv.get_fitted_params())
+    param_grid = optuna_param_grids[grid_key]
+    forecaster = forecasters_optuna_test[forecaster_key]
+
+    for sampler in optuna_samplers:  # Iterate over each sampler
```

Comment:
I think I'd have preferred `pytest.mark.parametrize` over for-loop for this. Other than knowing failing sampler name from the test name, it'd also ensure there is no fail fast to skip testing of next samplers.

Reply:
but `mark.parametrize` would break soft dependency isolation, since the samplers are `optuna` native objects.

It would work using pytest fixtures and request.params, but that might be a bit of a stretch for the payoff.

Reply:
I don't think so. If `optuna_samplers` fixture always return a list, either with actual samplers or with a list of a single value being None, I think it's enough. There will be no soft dependency issues.

But anyways, it's not a blocking request from my side. I'm happy with as it is.

Reply:
@yarnabrina unfortunatly @fkiraly is right, when using mark.parametrize it fails, even when using indirect=true tried that before.
But @fkiraly is also right about request.params. Adding something like 

`def optuna_samplers():
    if _check_soft_dependencies("optuna"):
        import optuna

        samplers = [
            None,
            optuna.samplers.NSGAIISampler(seed=42),
            optuna.samplers.QMCSampler(seed=42),
            optuna.samplers.CmaEsSampler(seed=42),
        ]
        return samplers


@pytest.fixture(scope="module", params=optuna_samplers())
def sampler(request):
    return request.param`

should work. So if you like I can try and implement it that way

Reply:
Done

Reply:
@yarnabrina thank you for your provided snipped. I think there was a missunderstanding from my side how you expected `mark.parametrize` and `fixture` decorator being used.
The implementation you suggested above works well. Succeeds with the 4 different provided samplers if optuna is installed and skipping the "None" sampler if optuna not installed.
So I took over the parts you suggested above.